outputmanager.cpp

```diff
@@ -16,3 +16,11 @@ OutputManager::OutputManager(QObject *parent) : QObject(parent)
 
 }
 
+void OutputManager::handleNewData(const QByteArray &data)
+{
+    // append raw data to the session buffer
+    buffer.append(data);
+
+    // notify that we have new data (in display format)
+    emit dataConverted(QString::fromLocal8Bit(data).remove('\n'));
```

Comment:
Ho due dubbi su questa riga.

Il primo è che `remove` toglie tutte le occorrenze della stringa specificata come argomento (http://doc.qt.io/qt-5/qstring.html#remove-2), ed il payload che arriva a questa funzione è stato recuperato tramite `readAll`, che torna tutti i dati in attesa di essere letti (http://doc.qt.io/qt-5/qiodevice.html#readAll).  Se quindi in attesa c'erano due linee, ciascuna delle quali con il proprio a capo alla fine, questa riga li rimuove entrambi e fa sì che in output tu mostri una singola riga collassata, es.

Input:
```
ciao
mondo
```

Output:
```
ciaomondo
```

Mi sembra che potenzialmente tu possa perdere informazione così facendo; magari però nella pratica succede che i dati vengono resi disponibili da seriale solo quando arriva un terminatore di linea.  Volevo capire se avevi considerato questo problema.

Il secondo è che qua assumi che il terminatore che ti arriva sia `\n`, ma quello che mandi tu in scrittura su seriale è `\r\n`.  In base a cosa hai fatto questa distinzione?